asistente_ladm_col/tests/test_quality_validations.py

```diff
@@ -395,6 +396,62 @@ def test_check_missing_survey_points_in_buildings(self):
         self.assertIn('Point (1091314.24563915398903191 1121448.97160633862949908)', geometries)
         self.assertIn('Point (1091213.62314918311312795 1121543.89559642062522471)', geometries)
 
+    def test_validate_topology_relation_between_point_boundary_boundary(self):
+        print('\nINFO: Validating that the relation between point boundary and boundary is registered in the topology table ...')
+
+        gpkg_path = get_test_copy_path('geopackage/tests_data.gpkg')
+        uri_boundary_points = gpkg_path + '|layername={layername}'.format(layername='good_boundary_points')
+        uri_boundary = gpkg_path + '|layername={layername}'.format(layername='good_boundary')
+        uri_pointsCcl_table = gpkg_path + '|layername={layername}'.format(layername='pointsCcl')
+
+        boundary_layer = QgsVectorLayer(uri_boundary, 'boundary_points', 'ogr')
+        boundary_points_layer = QgsVectorLayer(uri_boundary_points, 'boundary', 'ogr')
+        pointsCcl_table = QgsVectorLayer(uri_pointsCcl_table, 'pointsCcl', 'ogr')
+
+        # verify that the relation between point boundary and boundary is registered in the topology table
+        missing_topology = list()
+        duplicates_topology = list()
+        points_selected = self.qgis_utils.geometry.join_boundary_points_with_boundary_discard_nonmatching(boundary_points_layer, boundary_layer)
+
+        for point_selected in points_selected:
+            boundary_point_id = point_selected[ID_FIELD]
+            boundary_id = point_selected['{}_2'.format(ID_FIELD)]
+            expr = '"boundary_point_id" = {} and "boundary_id" = {}'.format(boundary_point_id, boundary_id)
+            it_select_features = pointsCcl_table.getFeatures(expr)
```

Comment:
We can optimize this by calling `pointsCcl_table.getFeatures()` only once, mapping the result into a dictionary like `my_dict = {"1501-9820": 2}` where 1501 is the `boundary_point_id`, 9820 is the `boundary_id` and 2 is the count of occurrences of such pair in `pointsCcl_table`.

In this way we just go to the DB once and build a dict where we can search easily the information we want.

Reply:
it was ajusted